drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -383,6 +385,7 @@ int stm32_i2c_msg_write(const struct device *dev, struct i2c_msg *msg,
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	struct i2c_stm32_data *data = DEV_DATA(dev);
 	I2C_TypeDef *i2c = cfg->i2c;
+	int is_timeout = 0;
```

Comment:
This could be `bool is_timeout = false;` and avoid some MISRA problems (treating `int` expressions as boolean).